diff --git a/utils/RawQueryFragment.js b/utils/RawQueryFragment.js
index 7358ac09760b129d2df01db9236377e2a5e0f8b7..86cf92fb4e9bab52d8c5f41f1a74556ba0e0060c 100644
--- a/utils/RawQueryFragment.js
+++ b/utils/RawQueryFragment.js
@@ -23,7 +23,11 @@ class RawQueryFragment {
         this.#key = `[raw]: ${this.sql} (#${RawQueryFragment.#index++})`;
     }
     as(alias) {
+      if (alias.startsWith('`') || alias.startsWith('"')) {
         return new RawQueryFragment(`${this.sql} as ${alias}`, this.params);
+      }
+
+      return new RawQueryFragment(`${this.sql} as ??`, [...this.params, alias]);
     }
     valueOf() {
         throw new Error(`Trying to modify raw SQL fragment: '${this.sql}'`);
